packages/typespec-ts/src/modular/buildProjectFiles.ts

```diff
@@ -242,12 +242,12 @@ export function emitPackage(
     packageInfo.scripts["unit-test"] =
       "npm run unit-test:node && npm run unit-test:browser";
     packageInfo.scripts["unit-test:node"] =
-      'mocha -r esm --require ts-node/register --reporter ../../../common/tools/mocha-multi-reporter.js --timeout 1200000 --full-trace "test/{,!(browser)/**/}*.spec.ts"';
+      'mocha --reporter ../../../common/tools/mocha-multi-reporter.js --timeout 1200000 --full-trace "test/{,!(browser)/**/}*.spec.ts"';
```

Comment:
have the same question here and the below.

Reply:
is the question here about `../../../common/tools/mocha-multi-reporter.js`? It is in old code too.  should it be removed too?

The reason to remove esm and ts-node/register requires is that for modular we will use the ts-node/esm loader

Reply:
I am a little curious what's the difference between the two ? what I might know ts-node/register is for commonjs, but I am kind of getting confused why put the esm and ts-node/register together before.
I would be really appreciate it if you can explain more about the background to me.

Reply:
'ts-node/register' transform .ts to .js but they are still in ESM.  For packages without "type": "module", js files are treated as commonjs, so we need 'esm' to transform ESM test files into commonjs